.github/workflows/vm-kernel.yaml

```diff
@@ -258,11 +258,33 @@ jobs:
           date="$(date -u '+%FT%TZ')"
           echo "suffix=-neon-$commit-$date" | tee -a $GITHUB_OUTPUT
 
+      - name: Load base image tags and SHAs into env
+        env:
+          MATRIX_ARCH: ${{ matrix.arch }}
+        run: |
+          case "$MATRIX_ARCH" in
+              'amd64' | 'arm64')
+                  ;;
+              *)
+                  echo "Unexpected architecture $MATRIX_ARCH"
+                  exit 1
+                  ;;
+          esac
+
+          arch="${MATRIX_ARCH@U}" # upper-case
+          grep -v '^#' versions.env | grep -E "^[A-Z_]+(_TAG|_SHA_$arch)=" | sed -e "s/_$arch=/=/g" >> $GITHUB_ENV
```

Comment:
We have that command repeated couple of times and the pattern is pretty complicated. 

Wdyt about extracting it to a helper shell script, to use it like `source $(export_versions.sh)` ?

Reply:
Yeah, makes sense. During development I had some issues because this entire step is repeated for a few workflows -- I'll see if I can extract it into a common step shared between workflows.

Reply:
Implemented in 964f72157459aec7a8d6c3aa6985b177425ed21c, 33d5de187efc590d23a1a528ff5f2102b69e77f6, 4397b075df566f2aca19cee96bbe2ce7d8d48894